types/api.d.ts

```diff
@@ -0,0 +1,49 @@
+import {Uri} from 'vscode';
+
+declare interface BasToolkit {
```

Comment:
I think adding in-depth documentation to everything here may be part of the scope
of this task as the goal of exporting the types "properly" from technical perspective
exists to serve the higher goal of having good type signatures that are easy to use.